lldb/source/Target/SwiftLanguageRuntimeNames.cpp

```diff
@@ -324,7 +324,7 @@ static lldb::ThreadPlanSP GetStepThroughTrampolinePlan(Thread &thread,
   return new_thread_plan_sp;
 }
 
-bool SwiftLanguageRuntime::IsSwiftMangledName(const char *name) {
+bool SwiftLanguageRuntime::IsSwiftMangledName(llvm::StringRef name) {
```

Comment:
The StringRef isn't necessarily NUL-terminated, and `swift::Demangle::isSwiftSymbol()` takes a `const char *`. To make this correct you'd need to round-trip to a std::string/SmallString to guarantee the terminator, which seems silly performance-wise.

Reply:
Adding @fredriss to the thread because we discussed this last week.

Reply:
Or are we going to hope that isSwiftSymbol only looks at the first two chars and that will never crash?

Reply:
Well, even looking at the first 2 chars can be going too far in the string... I hate the interface. Could we change `swift::Demangle::isSwiftSymbol()` to take a length in addition to the pointer or is this ABI?

Reply:
I think I can't follow. This is the implementation of isSwiftSymbol:
```
bool swift::Demangle::isSwiftSymbol(const char *mangledName) {
  StringRef mangledNameRef(mangledName);
  return isSwiftSymbol(mangledNameRef);
}
```
So if the const char* we have here is ever not null terminated it would have already exploded. This change is just moving the StringRef construction one call up.

Reply:
> Well, even looking at the first 2 chars can be going too far in the string... I hate the interface. Could we change `swift::Demangle::isSwiftSymbol()` to take a length in addition to the pointer or is this ABI?

I think that is the real answer. That is both faster (because it avoids building the stringref which scans the entire string) and more secure. We may need to keep the legacy interface around for ABI compatibility, but that's  for @mikeash ? to decide.

Reply:
Maybe I'm missing something, but we already have both:

```bool isSwiftSymbol(llvm::StringRef mangledName);
bool isSwiftSymbol(const char *mangledName);```

Changing this parameter should just make it call the other one. The `StringRef` variant already effectively takes a pointer and a length (since that's what a `StringRef` is, so you shouldn't need to add anything more.

Reply:
>  but we already have both

I completely missed that. Sorry!

Reply:
This LGTM then

Reply:
No problem! Glad it's sorted out.